src/daemon/distribution.c

```diff
@@ -336,3 +336,49 @@ double distribution_squared_deviation_sum(distribution_t *dist) {
   pthread_mutex_unlock(&dist->mutex);
   return squared_deviation_sum;
 }
+
+/* TODO(bkjg): add tests for this function */
+int distribution_sub(distribution_t *d1, distribution_t *d2) {
+  if (d1 == NULL || d2 == NULL) {
+    return EINVAL;
+  }
+
+  if (d1->num_buckets != d2->num_buckets) {
+    return EINVAL;
+  }
+
+  pthread_mutex_lock(&d1->mutex);
+  pthread_mutex_lock(&d2->mutex);
+
+  if (d1->total_sum < d2->total_sum) {
```

Comment:
I suppose it would be easier to consider that the first argument is always greater than the second one in order to avoid cases analysis and code duplication

Reply:
There is a possibility that the distribution will be reseted and this function should be able to handle this case. The first argument is the distribution that was updated more recently, however due to the reset, it still could have smaller counters.